ark/utils/segmentation_utils.py

```diff
@@ -127,3 +129,55 @@ def concatenate_csv(base_dir, csv_files, column_name="point", column_values=None
             combined_data = pd.concat((combined_data, temp_data), axis=0, ignore_index=True)
 
     combined_data.to_csv(os.path.join(base_dir, "combined_data.csv"), index=False)
+
+
+def visualize_watershed_transform(segmentation_labels_xr, channel_data_xr,
+                                  output_dir,
+                                  overlay_channels, fovs=None,
+                                  save_tifs='overlays'):
+    """Runs the watershed transform over a set of probability masks output by deepcell network
+    Saves xarray to output directory
+    Args:
+        segmentation_labels_xr (xarray.DataArray): xarray containing segmentation labels
+        channel_data_xr (xarray.DataArray): xarray containing TIFs
+        output_dir (str): path to directory where the output will be saved
+        overlay_channels (tuple): channels to overlay segmentation output over
+        fovs (list): field of view
+        save_tifs (str): flag to control what level of output to save. Must be one of:
+        all - saves all tifs
+        overlays - saves color overlays and segmentation masks
+    """
+
+    # error check model selected for local maxima finding in the image
+
+    # loop through all fovs and segment
+    if fovs is None:
+        fovs = segmentation_labels_xr.fovs
+    for fov in fovs:
+        # ignore low-contrast image warnings
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+
+            if save_tifs != 'none':
+                # save segmentation label map
+                for chan_list in overlay_channels:
+                    input_data = channel_data_xr.loc[fov, :, :, chan_list].values
+                    save_path = '_'.join([f'{fov.values}', *chan_list, 'overlay.tiff'])
+                    plot_utils.plot_overlay(
+                        segmentation_labels_xr.loc[fov, :, :, 'whole_cell'].values,
+                        plotting_tif=input_data,
+                        path=os.path.join(output_dir, save_path)
+                    )
+
+            if save_tifs == 'all':
+                channel = chan_list[0]
```

Comment:
isn't `chan_list` out of scope here (looks like it's outside of the loop over `overlay_channels`).